src/bindings/io/doc/io_doc.c

```diff
@@ -0,0 +1,365 @@
+	// someIoLibUpdateFd (operationData->handle, elektraIoFdGetFd (fdOp), elektraIoFdGetFlags (fdOp));
+
+	return 0;
+}
+
+/**
+ * Add file descriptor to IO-Binding
+ * @see kdbio.h ElektraIoInterface::addFd
+ */
+static int ioDocBindingAddFd (ElektraIoInterface * binding, ElektraIoFdOperation * fdOp)
+{
+	DocOperationData * operationData = newOperationData ();
+	if (operationData == NULL)
+	{
+		return -1;
+	}
+	// You can use private data stored in the IO binding
+	// e.g. MyData data = (MyData *)elektraIoBindingGetData (binding);
+
+	elektraIoFdSetBindingData (fdOp, operationData);
+	// You can store additional data for each operation in your operationData structure
+	operationData->operation.fd = fdOp;
+	operationData->bar = "foo";
+
+	// Here you need to add the operation to the I/O management library
+	// ioDocBindingFdCallback holds an example callback to pass to your I/O management library
+
+	// assume SomeIoLibHandle * someIoLibAddFd(int fd, int flags, int enabled, void * privateData, callback)
+	// operationData->handle = someIoLibAddFd(elektraIoFdGetFd (fdOp), elektraIoFdGetFlags (fdOp), elektraIoFdIsEnabled (fdOp), &fdOp,
+	// ioDocBindingFdCallback)
+
+	return 0;
+}
+
+/**
+ * Remove file descriptor from IO-Binding.
+ * @see kdbio.h ElektraIoInterface::removeFd
+ */
+static int ioDocBindingRemoveFd (ElektraIoFdOperation * fdOp)
+{
+	assert (elektraIoFdGetBindingData (fdOp) != NULL);
+	DocOperationData * operationData = (DocOperationData *)elektraIoFdGetBindingData (fdOp);
+
+	// Stop polling of the file descriptor and free data afterwards
+	// someIoLibStopFd (operationData->handle);
+	elektraFree (operationData);
+
+	return 0;
+}
+
+/**
+ * Update timer in IO-Binding.
+ * @see kdbio.h ElektraIoInterface::removeFd
+ */
+static int ioDocBindingUpdateTimer (ElektraIoTimerOperation * timerOp)
+{
+	assert (elektraIoTimerGetBindingData (timerOp) != NULL);
+	DocOperationData * operationData = (DocOperationData *)elektraIoTimerGetBindingData (timerOp);
+
+	// Based on elektraIoTimerIsEnabled (timerOp) enable or disable the operation
+	// Based on elektraIoTimerGetInterval (timerOp) change the timer interval
+	// someIoLibUpdateTimer (operationData->handle, elektraIoTimerGetInterval (timerOp), elektraIoTimerIsEnabled (timerOp));
+
+	return 0;
+}
+
+/**
+ * Add timer for IO-Binding.
+ * @see kdbio.h ElektraIoInterface::addTimer
+ */
+static int ioDocBindingAddTimer (ElektraIoInterface * binding, ElektraIoTimerOperation * timerOp)
+{
+	DocOperationData * operationData = newOperationData ();
+	if (operationData == NULL)
+	{
+		return -1;
+	}
+	// You can use private data stored in the IO binding
+	// e.g. MyData data = (MyData *)elektraIoBindingGetData (binding);
+
+	elektraIoTimerSetBindingData (timerOp, operationData);
+	// You can store additional data for each operation in your operationData structure
+	operationData->operation.timer = timerOp;
+
+	// Here you need to add the operation to the I/O management library
```

Comment:
These comments won't be part of Doxygen docu.

You can also include snippets of code (even of this file) using Doxygen's @snippet functionality.